Nodejs/Product/Nodejs/TypeScriptHelpers/TsConfigJsonFactory.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.NodejsTools.TypeScript
+{
+    internal static class TsConfigJsonFactory
+    {
+        public static async Task<TsConfigJson> CreateAsync(string fullPathToFile)
+        {
+            if (fullPathToFile == null || !fullPathToFile.EndsWith(NodejsConstants.TsConfigJsonFile, StringComparison.OrdinalIgnoreCase))
```

Comment:
>fullPathToFile.EndsWith [](start = 43, length = 23)

What about calling `Path.GetFilename` so that we don't have false positives like `mytsconfig.json`?

Reply:
Even better, call `IsTsConfigJsonFile`.

---
In reply to: [172942202](https://github.com/Microsoft/nodejstools/pull/1886#discussion_r172942202) [](ancestors = 172942202)